spring-cloud-huawei-servicecomb-discovery/src/main/java/com/huaweicloud/servicecomb/discovery/discovery/MicroserviceCache.java

```diff
@@ -35,19 +35,17 @@ public static void initService(List<Microservice> list) {
         list.forEach(item ->
                 item.getInstances().forEach(ins -> {
                     ins.setServiceName(item.getServiceName());
-                    ins.getEndpoints().forEach(ep -> {
-                        microserviceList.put(ep.replaceAll("rest://", ""), ins);
-                    });
+                    ins.getEndpoints()
+                        .forEach(ep -> microserviceList.put(ep.replaceAll("rest://", ""), ins));
                 })
         );
     }
 
     public static void initInsList(List<MicroserviceInstance> list, String serviceName) {
         list.forEach(ins -> {
             ins.setServiceName(serviceName);
-            ins.getEndpoints().forEach(ep -> {
-                microserviceList.put(ep.replaceAll("rest://", ""), ins);
-            });
+            ins.getEndpoints()
+                .forEach(ep -> microserviceList.put(ep.replaceAll("rest://", ""), ins));
```

Comment:
I am not sure where `getMicroserviceIns` is called. Seams use microservice instance id as the key is better. Not all endpoints starts with rest.

Reply:
Ribbon's  com.netflix.loadbalancer.Server choose the endpint as the instance id, so I use  endpoint.    
In spring cloud all endpoints with rest.